src/TBuffer.h

```diff
@@ -193,9 +193,13 @@ class TBuffer
 
 
     std::deque<TChar> bufferLine;
+    // stores the sequence of TChars that make up a line in a buffer
     std::deque<std::deque<TChar>> buffer;
-    QStringList timeBuffer;
+    // stores the actual content of lines
```

Comment:
This extensive comment now blurs lines between "what is the thing", "why do we use it", "how do we use it", "how does it relate to others", etc.
I concur we don't currently have a good place (in wiki maybe?) to document and discuss such general code architecture designs, may be worthwhile and more central location than having only some items and relations descriptions scattered throughout all Mudlet source code whenever a thing is used. RE: http://aosabook.org/

Reply:
Without spending too much energy on this as there's a lot of other unintuitive variables that could use a comment, how about this -

```cpp
// stores the actual content of lines as QChars. There is a 1:1 mapping between a letter
// and a TChar, except blank lines where there's no letter but a single TChar
```

Reply:
Leris rightfully pointed out that it's not clear what a TChar is, so this description really leads you on a goose chase. Perhaps this should be mentioned at a place where TChar is. For now lets keep the comment as-is because it's simple and easy to follow?

Reply:
Okay, keep the comment as you originally proposed. FTR (but continuing the potential repaint of the bicycle store) the use of the term `letter` in:
>// stores the actual content of lines as QChars. There is a 1:1 mapping between a letter
>// and a TChar, except blank lines where there's no letter but a single TChar

is confusingly wrong - a `QChar` is NOT a "letter" - but instead the unit/quantum of a UTF-16 encoded string - some graphemes are indeed a single `QChar` but many (those not in the Unicode Basic Multi-lingul Plane) are not.